### PR TITLE
Created the school library and implemented the file name

### PR DESCRIPTION
**Pull request by richie1988**
## I Created the class Person with the following:**

- Instance vars: @id, @name, and @age.
- Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
- Getters for @id, @name, and @age.
- Setters for @name and @age.
- Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.
- Public method can_use_services? that returns true if a person is of age or if they have permission from parents.

## Then I Created class Student with the following:**

- Inherits from Person.
- Constructor extends parent's constructor by adding @classroom and a parameter for it.
- Method play_hooky that returns "¯\(ツ)/¯".

## I Created the  class Teacher with the following:**

- Inherits from Person.
- Constructor extends parent's constructor by adding @specialization and a parameter for it.
- Override can_use_services? so it always returns true.
- Each class should be saved in a separate file.

### CODE REVIEW

File: library/person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(name: 'Unknown', age: 0, parent_permission: true)
```
Comment (zeff96) [inline library/person.rb]: - [ ] Hi there! 👋 Awesome job on the work so far! 🌟 Just a friendly reminder 🚨: according to the project guidelines, only 'name' and 'parent_permission' are optional, and they should have default values. 📋 Please make sure to always adhere to the project guidelines. 👍 Thanks a bunch! 🙌

Screenshot from 2023-11-13 19-47-15

```suggestion
  def initialize(age, name = "Unknown", parent_permission = true)
```

File: library/student.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Student < Person
+  attr_reader :classroom
+
+  def initialize(name: 'Unknown', age: 0, parent_permission: true, classroom: nil)
```
Comment (zeff96) [inline library/student.rb]: - [ ] Hi there! 👋 Awesome job on the work so far! 🌟 Just a friendly reminder 🚨: according to the project guidelines, only 'name' and 'parent_permission' are optional, and they should have default values. 📋 Please make sure to always adhere to the project guidelines. 👍 Thanks a bunch! 🙌

```suggestion
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: library/teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require_relative 'person'
+
+class Teacher < Person
+  attr_reader :specialization
+
+  def initialize(name: 'Unknown', age: 0, specialization: nil)
```
Comment (zeff96) [inline library/teacher.rb]: - [ ] Hi there! 👋 Awesome job on the work so far! 🌟 Just a friendly reminder 🚨: according to the project guidelines, only 'name' and 'parent_permission' are optional, and they should have default values. 📋 Please make sure to always adhere to the project guidelines. 👍 Thanks a bunch! 🙌

```suggestion
 def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,115 @@
...
+
+<!-- - GitHub: [@lily4178993](https://github.com/lily4178993)
+- Medium: [@nellytelli](https://medium.com/@nellytelli)
+- LinkedIn: [LinkedIn](https://linkedin.com/in/nellytelli) -->
+
+<!-- 👤 **Raphael Miyoyo** -->
+
+<!-- - GitHub: [@rmiyoyo](https://github.com/rmiyoyo)
+- Twitter: [@r_miyoyo](https://twitter.com/r_miyoyo)
+- LinkedIn: [LinkedIn](https://linkedin.com/in/raphael-miyoyo) -->
```
Comment (zeff96) [inline README.md]: - [ ] Hey! 👋 To give your README a more professional look, make sure to tidy up by removing any commented-out code. 🧹 Let's keep the content focused on information directly relevant to this project. 🚀 Thanks for your attention to detail! 🌟

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @richie1988 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct GitFlow



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — APPROVED** [review]
Hi @richie1988 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**